vendor/engines/projects/app/controllers/projects/projects_controller.rb

```diff
@@ -34,20 +36,34 @@ def update
       render action: :edit unless save_project
     end
 
+    def showing_archived?
+      params[:archived] == "true"
+    end
+    helper_method :showing_archived?
+
     private
 
     def project_params
       params.require(:projects_project).permit("active", "description", "name")
     end
 
+    def project_index_params
+      @project.active? ? {} : { archived: "true" }
+    end
+
     def save_project
       if @project.save
         flash[:notice] =
           I18n.t("controllers.projects.projects.#{action_name}.success", project_name: @project.name)
-        redirect_to facility_projects_path(@project.facility)
+        redirect_to facility_projects_path(@project.facility, project_index_params)
       end
     end
 
+    def filter_projects_for_index
+      @projects = showing_archived? ? @projects.inactive : @projects.active
```

Comment:
All this logic feels kind of convoluted. What about something like

```ruby
def index
  @active_projects = Project.active
  @inactive_project = Project.inactive
  @projects = showing_archived? @inactive_projects : @active_projects
end
```

I also feel the `@(in)active_project_count` instance variables don't add much. If we already have the actual relations, we could just call `.count` on it.